scripts/asm-installer/install_asm

```diff
@@ -1288,33 +1326,32 @@ is_rev_invalid() {
 }
 
 validate_ca_consistency() {
-  info "Checking existing CA..."
-  local INSTALLED_CA; INSTALLED_CA="$(kubectl -n istio-system get pod -l app=istio-ingressgateway \
-    -o jsonpath='{.items[].spec.containers[].env[?(@.name=="CA_ADDR")].value}')"
-
-  local CURRENT_CA; CURRENT_CA=citadel
+  local CURRENT_CA; CURRENT_CA="citadel"
   if is_meshca_installed; then
-    CURRENT_CA=mesh_ca
+    CURRENT_CA="mesh_ca"
+  elif is_privateca_installed; then
+    CURRENT_CA="private_ca"
   fi
+
   info "CA already in use: ${CURRENT_CA}"
 
-  if is_ca_citadel_to_meshca || is_ca_meshca_to_citadel; then
+  if [[ "${CA}" != "${CURRENT_CA}" ]]; then
     fatal "CA cannot be switched while performing upgrade. Please use ${CURRENT_CA} as the CA."
   fi
 }
 
-is_ca_citadel_to_meshca() {
-  ! is_meshca_installed && [[ "${CA}" = "mesh_ca" ]] && return 0
-}
-
-is_ca_meshca_to_citadel() {
-  is_meshca_installed && [[ "${CA}" = "citadel" ]] && return 0
-}
-
 is_meshca_installed() {
+  local INSTALLED_CA; INSTALLED_CA="$(kubectl -n istio-system get pod -l istio=ingressgateway \
+    -o jsonpath='{.items[].spec.containers[].env[?(@.name=="CA_ADDR")].value}')"
```

Comment:
Could you separate the "getting the installed CA" logic from the "check what CA it is" logic? something like

```
get_installed_ca() {
  kubectl -n istio-system get pod \
    -l istio=ingressgateway \
    -o jsonpath='{.items[].spec.containers[].env[?(@.name=="CA_ADDR")].value}'
}

is_privateca_installed() {
  if [[ "$(get_installed_ca)" != "ISTIOD_RA_CAS_API" ]]; then false; fi
}
```
And put "ISTIOD_RA_CAS_API" in a readonly variable at the top of the file.

Reply:
As of now, CA configuration is not a first class API - there is no common way routine to figure out which CA is installed. Each CA has its own method of detection. This will change in the near future, but that is why there are two separate functions for the different CA's

Reply:
You're right, I thought that the kubectl commands were the same but now I realize that the environment variables are different. My bad!